### PR TITLE
feat: resolve embedded languages automatically

### LINKED ISSUE #257: Error: No grammar provided for <insert-lang-here>
I'm trying to get async loading working properly for all languages supported by Shiki in [showcode.app](https://github.com/stevebauman/showcode/), but it seems some languages have issues and others do not. The issue is completely sporadic, even on the [Shiki playground](https://shiki-play.matsu.io/play) I receive this issue:

Screen Shot 2021-11-17 at 8 54 30 AM

Screen Shot 2021-11-17 at 8 57 54 AM

If you start clicking languages randomly, you'll eventually receive this error.

I haven't been able to track it down properly. My shiki `highlighter` instance is a singleton shared throughout my application, and when the languages load properly, they do stay loaded.

Can anyone point me in the direction of a potential fix? I've been digging through the source and trying to find potential issues, but my JS knowledge is certainly lacking.

Any help is greatly appreciated! ❤️  Thank you so much for this amazing tool.

### PR DESCRIPTION
**Pull request by muenzpraeger**
- [x] Add a test if possible
- [x] Format all commit messages with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

Fixes #257 

The current implementation hands off the ownership of understanding if and what embedded languages are needed to the document author.

```typescript
const highlighter = await getHighlighter({
  theme: 'nord',
  langs: ['html']
})
// Do we have JavaScript in that HTML? Ok, let's load it also.
await highlighter.loadLanguage('javascript');
// Do we have CSS in that HTML? Ok, let's load it also.
await highlighter.loadLanguage('css');
``` 

Now, this is a simple example. But there can be more complex ones, with multiple levels of embedded languages.

This PR ensures that every language that is embedded within another is auto-loaded, so that the implementers won't have to think about it.

The test shows the full example.

### CODE REVIEW

**Review by Gerrit0 — CHANGES_REQUESTED** [review]

File: packages/shiki/src/__tests__/embeddedLanguages.test.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { getHighlighter } from '../index'
+
+test('Requesting html as language also loads the embedded languages like JavaScript or CSS', async () => {
+  const highlighter = await getHighlighter({
+    theme: 'nord',
+    langs: ['html']
+  })
+  const languages = highlighter.getLoadedLanguages()
+  console.log(languages)
```
[the remark this answers is not in the document]
Comment (muenzpraeger) [reply packages/shiki/src/__tests__/embeddedLanguages.test.ts]: 🤦‍♂️

**Comment by orta** [comment]
Looks like some recursion from the import graph in the tests (cool PR though!)

**Comment by muenzpraeger** [comment]
Well, I should have run _all_ tests. 🤦‍♂️ Dang. Languages like `LaTeX` killed it.

I reverted the functionality only to the first level of embedded languages. That should be mostly sufficient, IMHO.

**Comment by orta** [comment]
Looks good to me 👍🏻

**Comment by orta** [comment]
Thanks!

**Pull request merged**